hibernate-core/src/main/java/org/hibernate/bytecode/internal/bytebuddy/ByteBuddyState.java

```diff
@@ -186,13 +186,25 @@ void clearState() {
 
 	private Class<?> load(Class<?> referenceClass, TypeCache<TypeCache.SimpleKey> cache,
 			TypeCache.SimpleKey cacheKey, Function<ByteBuddy, DynamicType.Builder<?>> makeProxyFunction) {
-		return cache.findOrInsert(
-				referenceClass.getClassLoader(),
-				cacheKey,
-				() -> make( makeProxyFunction.apply( byteBuddy ) )
-						.load( referenceClass.getClassLoader(), resolveClassLoadingStrategy( referenceClass ) )
-						.getLoaded(),
-				cache );
+
+		PrivilegedAction<Class<?>> delegateToPrivilegedAction =
+				new PrivilegedAction<Class<?>>() {
+
+					@Override
+					public Class<?> run() {
+						return cache.findOrInsert(
+								referenceClass.getClassLoader(),
+								cacheKey,
+								() -> make( makeProxyFunction.apply( byteBuddy ) )
+										.load( referenceClass.getClassLoader(), resolveClassLoadingStrategy(
+												referenceClass ) )
+										.getLoaded(),
+								cache
+						);
+					}
+				};
+		return SystemSecurityManager.isSecurityManagerEnabled()? AccessController.doPrivileged(delegateToPrivilegedAction) :
+				delegateToPrivilegedAction.run();
```

Comment:
```suggestion
		return cache.findOrInsert(
				referenceClass.getClassLoader(),
				cacheKey,
				() -> {
					PrivilegedAction<Class<?>> delegateToPrivilegedAction = new PrivilegedAction<Class<?>>() {

						@Override
						public Class<?> run() {
							return make( makeProxyFunction.apply( byteBuddy ) )
									.load( referenceClass.getClassLoader(), resolveClassLoadingStrategy( referenceClass ) )
									.getLoaded();
						}
					};
					return SystemSecurityManager.isSecurityManagerEnabled()? AccessController.doPrivileged(delegateToPrivilegedAction) :
				delegateToPrivilegedAction.run();
				},
				cache
		);
```

Reply:
@beikov your suggestion worked correctly, thank you!